src/components/PeerGroupsUpdate.tsx

```diff
@@ -0,0 +1,193 @@
+    const [peerGroups, setPeerGroups] = useState([] as GroupPeer[])
+    const [peerGroupsToSave, setPeerGroupsToSave] = useState({
+        ID: '',
+        groupsNoId: [],
+        groupsToSave: [],
+        groupsToRemove: [],
+        groupsToAdd: []
+    } as PeerGroupsToSave)
+
+    const [form] = Form.useForm()
+
+    useEffect(() => {
+        if (!peer) return
+        const gs = peer?.Groups?.map(g => ({ID: g?.ID || '', Name: g.Name} as GroupPeer)) as GroupPeer[]
+        const gs_name = gs?.map(g => g.Name) as string[]
+        setPeerGroups(gs)
+        setSelectedTagGroups(gs_name)
+        form.setFieldsValue({
+            groups: gs_name
+        })
+    }, [peer])
+
+    useEffect(() => {
+        setTagGroups(groups?.map(g => g.Name) || [])
+    }, [groups])
+
+    useEffect(() => {
+        const groupsToRemove = peerGroups.filter(pg => !selectedTagGroups.includes(pg.Name)).map(g => g.ID)
+        const groupsToAdd = (groups as Group[]).filter(g => selectedTagGroups.includes(g.Name) && !groupsToRemove.includes(g.ID || '') && !peerGroups.find(pg => pg.ID === g.ID)).map(g => g.ID) as string[]
+        const groupsNoId = selectedTagGroups.filter(stg => !groups.find(g => g.Name === stg))
+        setPeerGroupsToSave({
+            ...peerGroupsToSave,
+            ID: peer?.ID || '',
+            groupsToRemove,
+            groupsToAdd,
+            groupsNoId
+        })
+    }, [selectedTagGroups])
+
+    const tagRender = (props: CustomTagProps) => {
+        const { label, value, closable, onClose } = props;
+        const onPreventMouseDown = (event: React.MouseEvent<HTMLSpanElement>) => {
+            event.preventDefault();
+            event.stopPropagation();
+        };
+
+        return (
+            <Tag
+                color="blue"
+                onMouseDown={onPreventMouseDown}
+                closable={closable}
+                onClose={onClose}
+                style={{ marginRight: 3 }}
+            >
+                <strong>{value}</strong>
+            </Tag>
+        );
+    }
+
+    const optionRender = (label: string) => {
+        let peersCount = ''
+        const g = groups.find(_g => _g.Name === label)
+        if (g)  peersCount = ` - ${g.PeersCount || 0} ${(g.PeersCount && parseInt(g.PeersCount) > 1) ? 'peers' : 'peer'} `
+        return (
+            <>
+                <Tag
+                    color="blue"
+                    style={{ marginRight: 3 }}
+                >
+                    <strong>{label}</strong>
+                </Tag>
+                <span style={{fontSize: ".85em"}}>{peersCount}</span>
+            </>
+        )
+    }
+
+    const dropDownRender = (menu: React.ReactElement) => (
+        <>
+            {menu}
+            <Divider style={{ margin: '8px 0' }} />
+            <Row style={{padding: '0 8px 4px'}}>
+                <Col flex="auto">
+                    <span style={{color: "#9CA3AF"}}>Add new group by pressing "Enter"</span>
+                </Col>
```

Comment:
Codacy has a fix for the issue: [Replace `····························` with `··············`](https://app.codacy.com/gh/netbirdio/dashboard/pullRequest?prid=9560373)
```suggestion
              </Col>
```